lib/hamt.ml

```diff
@@ -513,6 +524,18 @@ module Make (Config : CONFIG) (Key : Hashtbl.HashedType) :
 
   let bindings hamt = fold (fun k v acc -> (k, v) :: acc) hamt []
 
+  let to_seq t = fold (fun k v acc () -> Seq.Cons ((k, v), acc)) t Seq.empty
```

Comment:
Is that as lazy as expected?

Reply:
It's as lazy as Stdlib.Map.to_seq. I think to have a properly lazy version we need an iterator that we can suspend/resume.

Reply:
I'm not sure about that. `Stdlib.Map.to_seq t` ultimately uses an explicit pattern matching on t (see `cons_enum`) to return immediately on the first found element. Your version uses `fold` that will always go through every element. 

I think we can make it actually lazy (ie. immediately build and return a sequence with the first element) by adapting the code of `fold`, as the Stdlib does for Map, arrays and lists.

Or am I wrong?

Reply:
Nah you're write. Don't know what I was thinking. Fixed now.